### PR TITLE
Issue/92

### PR DESCRIPTION
**Pull request by FRosner**

### CODE REVIEW

File: src/main/scala/de/frosner/dds/core/DDS.scala [hunk 1]
```diff
@@ -311,12 +311,18 @@ object DDS {
...
-      "The number of buckets parameter is optional having the default value of 10.",
-    parameters = "values: RDD[NumericValue], (optional) numBuckets: Int"
+      "The number of buckets parameter is optional - if omitted, Sturge's formula is used to determine an optimum number of bins",
+    parameters = "values: RDD[NumericValue], numBuckets: Integer"
   )
-  def histogram[N: ClassTag](values: RDD[N], numBuckets: Int = 100)(implicit num: Numeric[N]): Unit = {
-    val (buckets, frequencies) = values.map(v => num.toDouble(v)).histogram(numBuckets)
-    histogram(buckets, frequencies)
+  def histogram[N: ClassTag](values: RDD[N], numBuckets: Integer = null.asInstanceOf[Integer])(implicit num: Numeric[N]): Unit = {
+    if (numBuckets != null && numBuckets<2){
```
Comment (FRosner) [inline src/main/scala/de/frosner/dds/core/DDS.scala]: Please remove empty line

File: src/main/scala/de/frosner/dds/core/DDS.scala [hunk 2]
```diff
@@ -311,12 +311,18 @@ object DDS {
...
+      println("Number of Buckets must be greater than or equal to 2")
```
Comment (FRosner) [inline src/main/scala/de/frosner/dds/core/DDS.scala]: 'buckets' in small letters

File: src/main/scala/de/frosner/dds/core/DDS.scala [hunk 3]
```diff
@@ -642,4 +648,19 @@ object DDS {
     summarizeGroups(toBeGroupedValues.groupByKey())
   }
 
+  private[core] def lg2IntCeil(count:Long) : Integer = {
```
Comment (FRosner) [inline src/main/scala/de/frosner/dds/core/DDS.scala:651]: Does it make sense to put this into a Math util class somewhere? DDS.scala is becoming quite full. If so, can you create an issue to do it and give it the newbie label?

**Comment by FRosner** [comment]
@RPCMoritz looks quite good. I tried automatic rebasing but git screws up also in detecting the correct conflicts so I will do a manual merge and cherry pick the merge commit probably.

**Comment by FRosner** [comment]
@RPCMoritz let me know once you changed the few small things I complained about :smiling_imp:

File: src/main/scala/de/frosner/dds/core/DDS.scala (hunk 3 above)
Comment (RPCMoritz) [inline src/main/scala/de/frosner/dds/core/DDS.scala:651]: Unless you have other functions in mind to add to this class, I don't think it's worth the trouble yet.

**Comment by FRosner** [comment]
@RPCMoritz are you done? Let me know so I can try to merge the commits as soon as I have time :+1:

File: src/main/scala/de/frosner/dds/core/DDS.scala [hunk 4]
```diff
@@ -311,12 +311,18 @@ object DDS {
...
     longDescription = "Plots a histogram of a numerical RDD for the given number of buckets. " +
-      "The number of buckets parameter is optional having the default value of 10.",
-    parameters = "values: RDD[NumericValue], (optional) numBuckets: Int"
+      "The number of buckets parameter is optional - if omitted, Sturge's formula is used to determine an optimum number of bins",
+    parameters = "values: RDD[NumericValue], numBuckets: Integer"
   )
-  def histogram[N: ClassTag](values: RDD[N], numBuckets: Int = 100)(implicit num: Numeric[N]): Unit = {
-    val (buckets, frequencies) = values.map(v => num.toDouble(v)).histogram(numBuckets)
-    histogram(buckets, frequencies)
+  def histogram[N: ClassTag](values: RDD[N], numBuckets: Integer = null.asInstanceOf[Integer])(implicit num: Numeric[N]): Unit = {
```
Comment (FRosner) [inline src/main/scala/de/frosner/dds/core/DDS.scala]: @RPCMoritz While merging I just saw that you changed it to Integer. Why didn't you leave Int?

File: src/test/scala/de/frosner/dds/core/DDSTest.scala [hunk 5]
```diff
@@ -251,6 +251,16 @@ class DDSTest extends FlatSpec with Matchers with MockFactory with BeforeAndAfte
...
+    DDS.start(mockedServer)
+    val values = sc.makeRDD(List(0,5,15,3,8))
+    DDS.histogram(values)
+
+    val actualChart = mockedServer.lastServed.get.asInstanceOf[Histogram]
+    actualChart.bins.toList shouldBe List(0,3.75,7.5,11.25,15.0)
+    actualChart.frequencies.toList shouldBe List(2,1,1,1)
+  }
+
   "Correct pie chart from RDD after groupBy" should "be served when values are already grouped" in {
```
Comment (FRosner) [inline src/test/scala/de/frosner/dds/core/DDSTest.scala:264]: @RPCMoritz We need to add a test for an illegal number of buckets (< 2) as well. I will do it once I finished merging.

File: src/main/scala/de/frosner/dds/core/DDS.scala (hunk 4 above)
Comment (RPCMoritz) [inline src/main/scala/de/frosner/dds/core/DDS.scala]: I went for Java compatibility, where int won't be expected to be null. I see now that Int is also just another class, and not directly backed by java int, so I'll change it back.

File: src/test/scala/de/frosner/dds/core/DDSTest.scala (hunk 5 above)
Comment (RPCMoritz) [inline src/test/scala/de/frosner/dds/core/DDSTest.scala:264]: You're right - missed that one (Though there should have probably been one even before I added this feature ;) )

**Comment by RPCMoritz** [comment]
@FRosner: Took care of the final issue that I was aware of - Unless Travis complains, this is good to go.

File: src/main/scala/de/frosner/dds/core/DDS.scala (hunk 4 above)
Comment (FRosner) [inline src/main/scala/de/frosner/dds/core/DDS.scala]: null.asInstanceOf[Int] seems to be 0 or something...

Comment (FRosner) [inline src/main/scala/de/frosner/dds/core/DDS.scala]: I will either overload or go back to integer.

**Comment by FRosner** [comment]
@RPCMoritz I created https://github.com/FRosner/spawncamping-dds/pull/189/files with the rebased branch issue/92-2

I will include the missing test today and merge it back to master.

**Pull request closed without merging**